metrics-operator/controllers/common/providers/datadog/datadog.go

```diff
@@ -0,0 +1,79 @@
+package datadog
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/DataDog/datadog-api-client-go/v2/api/datadogV1"
+	"github.com/go-logr/logr"
+	metricsapi "github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha2"
+	"io"
+	"net/url"
+	"strconv"
+	"time"
+
+	//nolint:gci
+	"net/http" //nolint:gci
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type KeptnDataDogProvider struct {
+	Log        logr.Logger
+	HttpClient http.Client
+	K8sClient  client.Client
+}
+
+// EvaluateQuery fetches the SLI values from datadog provider
+func (d *KeptnDataDogProvider) EvaluateQuery(ctx context.Context, metric metricsapi.KeptnMetric, provider metricsapi.KeptnMetricsProvider) (string, []byte, error) {
+	ctx, cancel := context.WithTimeout(ctx, 20*time.Second)
+	defer cancel()
+
+	// Assumed default metric duration as 1 day(s)
```

Comment:
mm good point this is something we simplify also in the Prometheus provider...

Reply:
That's a valid point, I suggest creating a ticket to address this limitation

Reply:
we do have an issue https://github.com/keptn/lifecycle-toolkit/issues/859 to improve this, so I would be ok with this solution for now

Reply:
I creaded https://github.com/keptn/lifecycle-toolkit/issues/981